src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/RazorPagesOptionsExtensions.cs

```diff
@@ -170,6 +170,50 @@ public static RazorPagesOptions AuthorizeFolder(this RazorPagesOptions options,
         public static RazorPagesOptions AuthorizeFolder(this RazorPagesOptions options, string folderPath) =>
             AuthorizeFolder(options, folderPath, policy: string.Empty);
 
+        /// <summary>
+        /// Associates a friendly rotue with the page at the specified <paramref name="path"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="RazorPagesOptions"/>.</param>
+        /// <param name="path">The page path.</param>
+        /// <param name="routeUrl">The route to associate with the page.</param>
+        /// <returns>The <see cref="RazorPagesOptions"/>.</returns>
+        public static RazorPagesOptions MapPageRoute(this RazorPagesOptions options, string path, string routeUrl)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(path));
+            }
+
+            if (string.IsNullOrEmpty(routeUrl))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(routeUrl));
+            }
+
+            options.Conventions.Add(new PageConvention(path, model =>
+            {
+                // Use the route specified in MapPageRoute for outbound routing.
+                foreach (var selector in model.Selectors)
+                {
+                    selector.AttributeRouteModel.SuppressLinkGeneration = true;
```

Comment:
Essentially what we do with `/Index` and `/` routes, no? Should we call this out in the summary \ remarks of this method?